client.js

```diff
@@ -0,0 +1,254 @@
+'use strict';
+
+const { MTProto } = require('@mtproto/core');
+const {
+  addUsersToMessages,
+  getFormedMessagesFromUpdates,
+} = require('./helpers');
+
+module.exports = class TelegramClient {
+  constructor({ apiId, apiHash, test }) {
+    this.apiID = apiId;
+    this.apiHash = apiHash;
+    this.test = test;
+    this.channelMessageSubscribers = {};
+  }
+
+  async connect(authKeys) {
+    if (this.client) return;
+
+    return await new Promise((resolve, reject) => {
+      this.client = new MTProto({
+        api_id: this.apiID,
+        api_hash: this.apiHash,
+        test: this.test,
+        authKeys,
+      });
+
+      const socket = Object.values(this.client.rpcs)[0].transport.socket;
+
+      const onOpen = () => {
+        this.client.updates.on('updates', ({ updates, users }) => {
```

Comment:
do we really need to listen to updates when we connect? I thought we only wanted to listen to updates when we have at least one channelMessageSubscribers

Reply:
We start receiving updates as soon as a socket is opened, those updates may or may not be related to a channel. The code here checks if there are channel related updates, and if we are subscribed then is passes them to the subscribers.